lbrynet/lbrynet_daemon/LBRYDaemon.py

```diff
@@ -1889,21 +1907,44 @@ def jsonrpc_publish(self, p):
         Returns:
             Claim txid
         """
-        # start(self, name, file_path, bid, metadata, fee=None, sources=None):
+
         name = p['name']
         bid = p['bid']
         file_path = p['file_path']
         metadata = p['metadata']
+
+        def _set_address(address, currency):
+            log.info("Generated new address for key fee: " + str(address))
+            metadata['fee'][currency]['address'] = address
+            return defer.succeed(None)
+
+        def _delete_data(lbry_file):
+            txid = lbry_file.txid
+            d = self._delete_lbry_file(lbry_file, delete_file=False)
+            d.addCallback(lambda _: txid)
+            return d
+
+        if not self.pending_claim_checker.running:
+            self.pending_claim_checker.start(30)
+
+        d = self._resolve_name(name, force_refresh=True)
+        d.addErrback(lambda _: None)
+
         if 'fee' in p:
-            fee = p['fee']
-        else:
-            fee = None
+            metadata['fee'] = p['fee']
+            assert len(metadata['fee']) == 1, "Too many fees"
+            for c in metadata['fee']:
+                if 'address' not in metadata['fee'][c]:
+                    d.addCallback(lambda _: self.session.wallet.get_new_address())
+                    d.addCallback(lambda addr: _set_address(addr, c))
 
         pub = Publisher(self.session, self.lbry_file_manager, self.session.wallet)
-
-        d = pub.start(name, file_path, bid, metadata, fee)
-        d.addCallbacks(lambda msg: self._render_response(msg, OK_CODE),
-                       lambda err: self._render_response(err.getTraceback(), BAD_REQUEST))
+        d.addCallback(lambda _: self._get_lbry_file_by_uri(name))
+        d.addCallbacks(lambda l: None if not l else _delete_data(l), lambda _: None)
```

Comment:
Try to avoid having any logic inside the lambda functions in these callbacks.  Tracing through twisted callbacks is hard enough.  Only use the lambda function to change the order or arguments, etc.